# **TRABAJO PRÁCTICO — Laboratorio de Métodos Cuantitativos Aplicados en la Gestión**

### **Tema del informe: Plataforma Steam y desempeño comercial de videojuegos**


#### **Integrantes del grupo**

*   Martina Bisio; registro: 921302

*   Victoria Carolina Hartmann; registro: 921287

*  Camila Belen Ortiz Espinoza; registro: 921297

* Rocio Luján Galván; registro: 921292

* Katherine Yesebet Freites Villalobos; registro: 921314

#### **Objetivo general**

Analizar qué características influyen en el desempeño comercial y la aceptación de videojuegos dentro de la plataforma Steam utilizando técnicas de análisis de datos, visualización y optimización cuantitativa.

#### **Uso de asistencia de IA**
Para la etapa inicial de exploración conceptual del dataset y definición de variables relevantes, se utilizó asistencia de IA mediante ChatGPT (modelo GPT-5.5).

#### **Prompt utilizado**

> Actuá como un analista de datos senior. Y usando todos los archivos .csv de Steam: steam.csv, steamspy_tag_data.csv, steam_media_data.csv, steam_description_data.csv, steam_requirements_data.csv y steam_support_info.csv. Tu tarea es:

> (1) identificar las variables más relevantes para responder “¿Qué características influyen en el desempeño comercial y aceptación de videojuegos dentro de Steam?”,

> (2) sugerir transformaciones útiles,

> (3) proponer qué tipo de gráfico comunica mejor cada hallazgo, y

> (4) señalar limitaciones del dataset que puedan afectar las conclusiones.
Respondé en español y justificá cada decisión.

#### **Análisis exploratorio conceptual del dataset**

Perfecto.
Pensándolo como un analista de datos senior y viendo la estructura COMPLETA de los `.csv`, ustedes ya tienen material suficiente para construir un análisis muy sólido tanto desde lo estadístico como desde lo económico-organizacional.

La pregunta central:

# “¿Qué características influyen en el desempeño comercial y aceptación de videojuegos dentro de Steam?”

abarca estos puntos:

* permite análisis cuantitativos.
* análisis temporales.
* interpretación económica.
* visualización.
* incluso optimización.

Y además conecta directamente con los datos disponibles.

---

# 1. VARIABLES MÁS RELEVANTES DEL DATASET

Voy a dividirlas según qué dimensión ayudan a explicar.

---

# A) Variables de desempeño comercial

Estas ayudan a medir “éxito económico”.

## Desde `steam.csv`

| Variable           | Justificación                                     |
| ------------------ | ------------------------------------------------- |
| `owners`           | Proxy principal de ventas/cantidad de compradores |
| `price`            | Permite analizar elasticidad/precio óptimo        |
| `positive_ratings` | Relación entre aceptación y éxito comercial       |
| `negative_ratings` | Impacto de mala recepción                         |
| `average_playtime` | Retención y engagement                            |
| `median_playtime`  | Complementa evitando distorsión por extremos      |
| `release_date`     | Análisis temporal/evolución histórica             |
| `genres`           | Detectar géneros más exitosos                     |
| `categories`       | Multiplayer, coop, achievements, etc.             |
| `developer`        | Concentración de mercado                          |
| `publisher`        | Diferencia indie vs grandes publishers            |
| `achievements`     | Contenido/recompensa al jugador                   |

---

# B) Variables de aceptación del usuario

Estas ayudan a medir “satisfacción”.

## Variables derivadas importantes

### ✔ Approval Ratio

Muy importante.


$$
approval\_ratio =
\frac{positive\_ratings}
{positive\_ratings + negative\_ratings}
$$


Justificación:

* estandariza aceptación,
* evita comparar solo volumen bruto,
* permite comparar juegos chicos y grandes.

---

### ✔ Engagement Ratio

$$
engagement =
\frac{average\_playtime}
{owners}
$$

Sirve para detectar:

* juegos muy retenedores,
* nichos fuertes,
* fidelización.

---

# C) Variables técnicas

## Desde `steam_requirements_data.csv`

| Variable                  | Utilidad              |
| ------------------------- | --------------------- |
| requisitos mínimos        | accesibilidad técnica |
| requisitos recomendados   | barrera de entrada    |
| soporte Windows/Mac/Linux | expansión de mercado  |

---

# D) Variables descriptivas y de contenido

## Desde `steam_description_data.csv`

| Variable          | Utilidad                           |
| ----------------- | ---------------------------------- |
| descripción corta | NLP futuro                         |
| descripción larga | análisis de marketing/comunicación |

---

# E) Variables de tags

## Desde `steamspy_tag_data.csv`

Este archivo resulta especialmente relevante para el análisis debido a la granularidad temática que aporta mediante las etiquetas asociadas a cada videojuego.

Porque permite:

* medir popularidad por etiquetas,
* análisis de nichos,
* saturación de mercado,
* tendencias históricas.

Ejemplos:

* Indie
* RPG
* Survival
* Multiplayer
* Anime
* Strategy

---

# F) Variables multimedia

## Desde `steam_media_data.csv`

| Variable                | Utilidad            |
| ----------------------- | ------------------- |
| cantidad de screenshots | marketing visual    |
| trailers/videos         | impacto promocional |

---

# G) Variables de soporte

## Desde `steam_support_info.csv`

| Variable      | Utilidad           |
| ------------- | ------------------ |
| website       | profesionalización |
| support_email | soporte formal     |

Puede servir para:

* diferenciar estudios profesionales vs pequeños,
* medir madurez organizacional.

---

# 2. TRANSFORMACIONES ÚTILES

Esta parte es CLAVE para el TP.

---

# A) Transformar `owners`

Problema:
viene como rango textual.

Ejemplo:

```python
"20000-50000"
```

## ✔ Transformación recomendada:

usar punto medio.

Ejemplo:

$$
owners\_mid = \frac{20000 + 50000}{2}
$$

Justificación:
permite:

* correlaciones,
* regresiones,
* métricas económicas,
* optimización.

---

# B) Crear variable “Indie”

```python
df['is_indie'] =
df['genres'].str.contains('Indie')
```

Justificación:
permite validar:

* crecimiento indie,
* diferencias de mercado,
* hipótesis temporal.

---

# C) Crear décadas/etapas históricas

La división temporal por etapas históricas permite contextualizar los cambios estructurales de la industria.

Ejemplo:

| Etapa             | Período   |
| ----------------- | --------- |
| Online temprano   | 2000–2006 |
| HD/Casual         | 2006–2013 |
| Streaming/Esports | 2013–2019 |

Justificación:

* análisis evolutivo,
* cambios estructurales,
* comparación histórica.

---

# D) Log-transform en ventas

Las ventas tienen distribución MUY sesgada.

Conviene:

$$
\log(owners\_mid + 1)
$$

Justificación:

* evitar dominancia de outliers,
* gráficos más legibles,
* correlaciones más estables.

---

# E) Revenue estimado

$$
estimated\_revenue = owners\_mid \times price
$$

IMPORTANTÍSIMO para:

* desempeño comercial,
* optimización,
* análisis económico.

---

# F) Clasificar publishers

Separar:

* grandes estudios,
* medianos,
* indies.

Puede hacerse por:

* cantidad de juegos,
* promedio de owners,
* revenue estimado.

---

# 3. QUÉ GRÁFICO USAR Y POR QUÉ

---

# A) Evolución histórica Indie vs No Indie

## ✔ Gráfico:

líneas temporales.

### Porque:

* muestra tendencias,
* permite comparar crecimiento,
* comunica evolución claramente.

---

# B) Precio vs Approval Ratio

## ✔ Scatterplot

Variables:

* X = precio
* Y = approval_ratio

### Porque:

permite ver:

* elasticidad percibida,
* tolerancia del consumidor,
* clustering.

---

# C) Géneros más exitosos

## ✔ Barplot horizontal

Variables:

* género
* owners/revenue promedio

### Porque:

facilita comparación categórica.

---

# D) Distribución de precios

## ✔ Histograma

Porque:

* muestra concentración,
* free-to-play,
* rango dominante.

---

# E) Ratings positivos vs negativos

## ✔ Scatterplot con escala log

Porque:

* evita distorsión,
* detecta juegos polémicos,
* muestra aceptación masiva.

---

# F) Correlaciones generales

## ✔ Heatmap

Variables:

* price
* playtime
* ratings
* achievements
* owners

Porque:

* resume relaciones rápidamente.

---

# G) Playtime por género

## ✔ Boxplot

Porque:

* muestra dispersión,
* outliers,
* diferencias reales de engagement.

---

# 4. LIMITACIONES IMPORTANTES DEL DATASET

ESTA PARTE ES FUNDAMENTAL para parecer un análisis serio.

---

# A) El dataset termina en 2019

Impacto:

* no captura:

  * pandemia,
  * explosión masiva indie reciente,
  * Steam Deck,
  * cambios post-2020.

Pero:
✔ sigue siendo válido históricamente.

---

# B) `owners` es estimado

No son ventas oficiales exactas.

Impacto:

* introduce incertidumbre,
* afecta revenue estimado.

---

# C) Falta información financiera real

No hay:

* costos de desarrollo,
* presupuesto marketing,
* ganancias netas.

Entonces:
❌ no puede medirse rentabilidad real.

---

# D) Sesgo hacia juegos populares

Muchos juegos chicos tienen:

* pocos ratings,
* datos incompletos,
* poca visibilidad.

---

# E) Correlación ≠ causalidad

Ejemplo:

* que un juego tenga achievements NO implica que venda más.
  Puede ser:
* juegos exitosos reciben más contenido.

---

# F) Tags y géneros superpuestos

Un juego puede ser:

* Indie
* RPG
* Survival
* Multiplayer

al mismo tiempo.

Entonces:

* hay multicolinealidad conceptual.

---

# CONCLUSIÓN ESTRATÉGICA

Viendo TODOS los `.csv`, el TP tiene muchísimo potencial porque combina:

✅ análisis temporal

✅ comportamiento del consumidor

✅ economía digital

✅ mercado de videojuegos

✅ engagement

✅ aceptación del usuario

✅ optimización

✅ análisis organizacional




## 1. Carga y exploración inicial de los datasets

En esta etapa se realizará la carga de los archivos `.csv` correspondientes a la plataforma Steam utilizando la librería Pandas.

El objetivo inicial consiste en:

- explorar la estructura general de cada dataset,
- identificar dimensiones y tipos de datos,
- evaluar la calidad de la información,
- detectar valores nulos o registros duplicados,
- y comprender el alcance temporal de los datos.

Este análisis preliminar resulta fundamental para garantizar que las transformaciones y análisis posteriores se realicen sobre información consistente y confiable.

In [ ]:
# Librerías principales
import pandas as pd
import numpy as np

# Visualización
import matplotlib.pyplot as plt
import seaborn as sns

## 1.1 Carga de los archivos

Se cargan los seis datasets principales relacionados con videojuegos dentro de la plataforma Steam.

Cada archivo contiene información distinta:

- `steam.csv`: información general de videojuegos
- `steamspy_tag_data.csv`: etiquetas y géneros asociados
- `steam_media_data.csv`: contenido multimedia
- `steam_description_data.csv`: descripciones de juegos
- `steam_requirements_data.csv`: requisitos técnicos
- `steam_support_info.csv`: información de soporte y contacto

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
# Carga de datasets

df_steam = pd.read_csv('/content/drive/MyDrive/archive data/steam.csv')

df_tags = pd.read_csv('/content/drive/MyDrive/archive data/steamspy_tag_data.csv')

df_media = pd.read_csv('/content/drive/MyDrive/archive data/steam_media_data.csv')

df_description = pd.read_csv('/content/drive/MyDrive/archive data/steam_description_data.csv')

df_requirements = pd.read_csv('/content/drive/MyDrive/archive data/steam_requirements_data.csv')

df_support = pd.read_csv('/content/drive/MyDrive/archive data/steam_support_info.csv')

## 1.2 Visualización inicial

Se muestran las primeras y últimas filas de cada dataset para comprender:

- estructura,
- variables disponibles,
- formato de los registros,
- y tipo de información almacenada.

In [ ]:
# Primeras filas

print("STEAM")
print(df_steam.head())

print("\nTAGS")
print(df_tags.head())

print("\nMEDIA")
print(df_media.head())

print("\nDESCRIPTION")
print(df_description.head())

print("\nREQUIREMENTS")
print(df_requirements.head())

print("\nSUPPORT")
print(df_support.head())

STEAM
   appid                       name release_date  english         developer  \
0     10             Counter-Strike   2000-11-01        1             Valve   
1     20      Team Fortress Classic   1999-04-01        1             Valve   
2     30              Day of Defeat   2003-05-01        1             Valve   
3     40         Deathmatch Classic   2001-06-01        1             Valve   
4     50  Half-Life: Opposing Force   1999-11-01        1  Gearbox Software   

  publisher          platforms  required_age  \
0     Valve  windows;mac;linux             0   
1     Valve  windows;mac;linux             0   
2     Valve  windows;mac;linux             0   
3     Valve  windows;mac;linux             0   
4     Valve  windows;mac;linux             0   

                                          categories  genres  \
0  Multi-player;Online Multi-Player;Local Multi-P...  Action   
1  Multi-player;Online Multi-Player;Local Multi-P...  Action   
2              Multi-player;Valve Anti

In [ ]:
# Últimas filas

print("STEAM")
print(df_steam.tail())

print("\nTAGS")
print(df_tags.tail())

print("\nMEDIA")
print(df_media.tail())

print("\nDESCRIPTION")
print(df_description.tail())

print("\nREQUIREMENTS")
print(df_requirements.tail())

print("\nSUPPORT")
print(df_support.tail())

STEAM
         appid                        name release_date  english  \
27070  1065230             Room of Pandora   2019-04-24        1   
27071  1065570                   Cyber Gun   2019-04-23        1   
27072  1065650            Super Star Blast   2019-04-24        1   
27073  1066700  New Yankee 7: Deer Hunters   2019-04-17        1   
27074  1069460                   Rune Lord   2019-04-24        1   

                developer             publisher    platforms  required_age  \
27070         SHEN JIAWEI           SHEN JIAWEI      windows             0   
27071      Semyon Maximov      BekkerDev Studio      windows             0   
27072         EntwicklerX           EntwicklerX      windows             0   
27073  Yustas Game Studio  Alawar Entertainment  windows;mac             0   
27074    Adept Studios GD  Alawar Entertainment  windows;mac             0   

                                              categories  \
27070                   Single-player;Steam Achievements

## 1.3 Estructura de los datasets

A continuación se analizarán:

- dimensiones,
- nombres de columnas,
- tipos de datos,
- y cobertura temporal.

Esto permite identificar qué variables serán útiles para el análisis económico y comercial posterior.

In [ ]:
# Dimensiones de cada dataset

print("Dimensiones STEAM:", df_steam.shape)
print("Dimensiones TAGS:", df_tags.shape)
print("Dimensiones MEDIA:", df_media.shape)
print("Dimensiones DESCRIPTION:", df_description.shape)
print("Dimensiones REQUIREMENTS:", df_requirements.shape)
print("Dimensiones SUPPORT:", df_support.shape)

Dimensiones STEAM: (27075, 18)
Dimensiones TAGS: (29022, 372)
Dimensiones MEDIA: (27332, 5)
Dimensiones DESCRIPTION: (27334, 4)
Dimensiones REQUIREMENTS: (27319, 6)
Dimensiones SUPPORT: (27136, 4)


In [ ]:
# Columnas y tipos de datos

print("Tipos de datos - STEAM")
print(df_steam.dtypes)

print("\nTipos de datos - TAGS")
print(df_tags.dtypes)

print("\nTipos de datos - MEDIA")
print(df_media.dtypes)

print("\nTipos de datos - DESCRIPTION")
print(df_description.dtypes)

print("\nTipos de datos - REQUIREMENTS")
print(df_requirements.dtypes)

print("\nTipos de datos - SUPPORT")
print(df_support.dtypes)

Tipos de datos - STEAM
appid                 int64
name                 object
release_date         object
english               int64
developer            object
publisher            object
platforms            object
required_age          int64
categories           object
genres               object
steamspy_tags        object
achievements          int64
positive_ratings      int64
negative_ratings      int64
average_playtime      int64
median_playtime       int64
owners               object
price               float64
dtype: object

Tipos de datos - TAGS
appid           int64
1980s           int64
1990s           int64
2.5d            int64
2d              int64
                ...  
world_war_i     int64
world_war_ii    int64
wrestling       int64
zombies         int64
e_sports        int64
Length: 372, dtype: object

Tipos de datos - MEDIA
steam_appid      int64
header_image    object
screenshots     object
background      object
movies          object
dtype: object

Tipos de dato

In [ ]:
# Conversión de fechas

df_steam['release_date'] = pd.to_datetime(df_steam['release_date'])

# Cobertura temporal

print("Fecha mínima:", df_steam['release_date'].min())

print("Fecha máxima:", df_steam['release_date'].max())

Fecha mínima: 1997-06-30 00:00:00
Fecha máxima: 2019-05-01 00:00:00


## 1.4 Calidad de los datos

Se evaluará la calidad general de los datasets mediante:

- detección de valores nulos,
- identificación de registros duplicados,
- y estadísticas descriptivas.

Este paso resulta importante para detectar posibles limitaciones metodológicas antes de realizar análisis más complejos.

In [ ]:
# Valores nulos por dataset

print("Nulos STEAM")
print(df_steam.isnull().sum())

print("\nNulos TAGS")
print(df_tags.isnull().sum())

print("\nNulos MEDIA")
print(df_media.isnull().sum())

print("\nNulos DESCRIPTION")
print(df_description.isnull().sum())

print("\nNulos REQUIREMENTS")
print(df_requirements.isnull().sum())

print("\nNulos SUPPORT")
print(df_support.isnull().sum())

Nulos STEAM
appid                0
name                 0
release_date         0
english              0
developer            1
publisher           14
platforms            0
required_age         0
categories           0
genres               0
steamspy_tags        0
achievements         0
positive_ratings     0
negative_ratings     0
average_playtime     0
median_playtime      0
owners               0
price                0
dtype: int64

Nulos TAGS
appid           0
1980s           0
1990s           0
2.5d            0
2d              0
               ..
world_war_i     0
world_war_ii    0
wrestling       0
zombies         0
e_sports        0
Length: 372, dtype: int64

Nulos MEDIA
steam_appid        0
header_image       0
screenshots        0
background         0
movies          1691
dtype: int64

Nulos DESCRIPTION
steam_appid             0
detailed_description    0
about_the_game          0
short_description       0
dtype: int64

Nulos REQUIREMENTS
steam_appid               0
pc_require

In [ ]:
# Registros duplicados

print("Duplicados STEAM:", df_steam.duplicated().sum())

print("Duplicados TAGS:", df_tags.duplicated().sum())

print("Duplicados MEDIA:", df_media.duplicated().sum())

print("Duplicados DESCRIPTION:", df_description.duplicated().sum())

print("Duplicados REQUIREMENTS:", df_requirements.duplicated().sum())

print("Duplicados SUPPORT:", df_support.duplicated().sum())

Duplicados STEAM: 0
Duplicados TAGS: 0
Duplicados MEDIA: 0
Duplicados DESCRIPTION: 0
Duplicados REQUIREMENTS: 0
Duplicados SUPPORT: 0


In [ ]:
# Estadísticas descriptivas

print("Estadísticas STEAM")
print(df_steam.describe())

print("\nEstadísticas TAGS")
print(df_tags.describe())

print("\nEstadísticas MEDIA")
print(df_media.describe())

print("\nEstadísticas REQUIREMENTS")
print(df_requirements.describe())

Estadísticas STEAM
              appid                   release_date       english  \
count  2.707500e+04                          27075  27075.000000   
mean   5.962035e+05  2016-12-31 14:21:17.252077568      0.981127   
min    1.000000e+01            1997-06-30 00:00:00      0.000000   
25%    4.012300e+05            2016-04-04 00:00:00      1.000000   
50%    5.990700e+05            2017-08-08 00:00:00      1.000000   
75%    7.987600e+05            2018-06-06 12:00:00      1.000000   
max    1.069460e+06            2019-05-01 00:00:00      1.000000   
std    2.508942e+05                            NaN      0.136081   

       required_age  achievements  positive_ratings  negative_ratings  \
count  27075.000000  27075.000000      2.707500e+04      27075.000000   
mean       0.354903     45.248864      1.000559e+03        211.027147   
min        0.000000      0.000000      0.000000e+00          0.000000   
25%        0.000000      0.000000      6.000000e+00          2.000000   
50%

## 1.5 Hallazgos preliminares

A partir de la exploración inicial pueden identificarse algunos aspectos relevantes:

- El dataset principal (`steam.csv`) concentra la mayor parte de las variables económicas y comerciales relevantes para el análisis.
- Existen variables categóricas y textuales que requerirán procesamiento previo para poder incorporarse a análisis cuantitativos.
- Algunas columnas presentan valores nulos, especialmente en datasets secundarios relacionados con soporte técnico o requisitos específicos.
- La cobertura temporal llega aproximadamente hasta el año 2019, por lo que las conclusiones deben interpretarse dentro de dicho contexto histórico.
- La presencia de variables como precio, cantidad estimada de propietarios, ratings positivos y tiempo de juego permite construir métricas económicas y de engagement relevantes para el trabajo.

En etapas posteriores se realizarán transformaciones y unificación de datasets para profundizar el análisis del desempeño comercial de videojuegos dentro de Steam.

## 2.A. Transformación de la Variable 'owners' (Punto Medio)

**Problema identificado:** En la columna original de Owners recita la cantidad promedio de usuarios que poseen el videojuego mediante la codificacion de rango de texto (dando el ejemplo de "20000-50000"). Siendo un string, no hay posibilades para analizar las operaciones matematicas correspodientes.

**Solución aplicada:** La aplicacion de una transformacion numerica que esta basada en el "punto medio del rango". Al extender tanto el limite superior y el inferior del texto en cada juego, calculamos el promedio numerico creando la variable llamada `owners_mid`.

La fórmula matemática aplicada para cada registro es:

$$owners\_mid = \frac{\text{Límite Inferior} + \text{Límite Superior}}{2}$$

**Justificación técnica y económica:** Esta transformación nos permite tratar la variable de forma continua para:
* Evaluar **correlaciones** (ej: relación entre el precio y el volumen de usuarios).
* Desarrollar **regresiones** y proyecciones de demanda.
* Calcular **métricas económicas** aproximadas (como la estimación de ingresos brutos por juego).

In [ ]:
#Utilizando IA para correguir la definicion, la usamos para convertir a enteros los valores
def calcular_punto_medio_owners(rango_texto):
    """
    Toma un string con formato 'MIN-MAX' (ej: '20000-50000'),
    separa los valores, los convierte a enteros y devuelve su promedio.
    """
    try:
        # Utilizando el guion como un separador de texto
        limites = str(rango_texto).split('-')

        # Cambiar el rango a un entero
        limite_inferior = int(limites[0].strip())
        limite_superior = int(limites[1].strip())

        # El calculo del punto medio
        punto_medio = (limite_inferior + limite_superior) / 2
        return punto_medio
    except Exception as e:
        # Devolver NaN a cada valor nulo o formato inesperados
        return np.nan

#Para aplicar la nueva columna numerica
df_steam['owners_mid'] = df_steam['owners'].apply(calcular_punto_medio_owners)

#Y su verificacion rapida
print("Muestra de la transformación de 'owners':")
print(df_steam[['name', 'owners', 'owners_mid']].head())

Muestra de la transformación de 'owners':
                        name             owners  owners_mid
0             Counter-Strike  10000000-20000000  15000000.0
1      Team Fortress Classic   5000000-10000000   7500000.0
2              Day of Defeat   5000000-10000000   7500000.0
3         Deathmatch Classic   5000000-10000000   7500000.0
4  Half-Life: Opposing Force   5000000-10000000   7500000.0


### 📊 Interpretación Económica e Implicancias del Negocio

En el analisis de *two-sided markets*, la cantidad de los usuarios acumulados del activo estrategico y la barrera de la entrada critica. Con el dataset que provee los rangos de trasacciones por los motivos de confidencialidad comercial, la variable `owners_mid` funciona como **proxy de la demanda cuantitativa histórica (Q)** para cada firma dentro del ecosistema de Steam.

Observando la salida de codigo, los resultados reflejan el comportamiento mencionado. Titulos pioneros de Valve como es *Counter-Strike* que registran la demanda de **15,000,000 de usuarios**, al mismo tiempo los clásicos como *Team Fortress Classic* o *Day of Defeat* se estabilizan en **7,500,000**. Disponiendo los valores continuos donde permite modelar la aproximacion de la **Función de Ingresos Totales (IT)** mediante:

$$\text{Ingreso Estimado} \approx \text{price} \times owners\_mid$$

**Nota metodológica del equipo:** El equipo pudo reconocer que el indicador representa el *ingreso devengado bruto potencial* en la lista de precios actuales. La realidad de las organizaciones digitales donde el valor no deduce la tasa de intermediación de la plataforma (la comisión del 30% de Valve), donde los efectos de la dicriminacion de los precios (oferta estacionales) o las elasticidades de los precios regional tampoco.

Sin embargo, este valor analitico de la transformacion se encuentra en la capacidad para **homogeneizar y estandarizar la escala de las firmas**. Donde pasar de rangos categóricos como `"10000000-20000000"` a métricas numéricas exactas, es uno de los pasos indispernsable para el calculo de las cuotas del mercado que se evalua la concentracion de la oferta respecto a la demanda y la clasificacion del exito comercial en las secciones posteriores.

## 2.B. Creación de Variable Booleana 'is_indie'

**Problema identificado:** Con la columna `genres` donde contiene variedad de etiquetas de texto (por ejemplo, `"Action;Casual;Indie"`). Con el fin de analizar el comportamiento de los desarrolladores indepediente (indies), se requiere aislar la carracteristica en una variable clara y binaria

**Solución aplicada:** Necesitamos crear la variable `dummy` o la indicadora con el nombre de `is_indie`. Con el uso de metodos de cadenas en pandas, se evaluara si la palabra `"Indie"` se preseta en el listado de generos

**Justificación técnica y económica:** Disponer de esta variable booleana (`True` o `False`) nos habilita a realizar:
**Estudios de crecimiento:** Medir la evolución del volumen de lanzamientos independientes año a año.
**Diferencias de mercado:** Comparar las estructuras de precios y niveles de recaudación entre el sector Indie y el corporativo.
**Validación de hipótesis temporales:** Analizar si el "boom" de los videojuegos independientes ha alterado la concentración de la demanda en la plataforma a lo largo del tiempo.

In [ ]:
# Con el apoyo de la IA para la correccion de errores presentados.
# 1. Al crear la variable booleana con el fin de evaluar la presencia de 'Indie' en la columna de generos
# Con el uso de na=False como medida de seguridad en el caso que haya valores nulos
df_steam['is_indie'] = df_steam['genres'].str.contains('Indie', na=False)

# 2. La verificacion de la proporcion del mercado en la muestra aleatoria
print(f"Porcentaje de juegos clasificados como Indie: {df_steam['is_indie'].mean() * 100:.2f}%")
print("-" * 50)
print("Muestra de verificación:")
print(df_steam[['name', 'genres', 'is_indie']].sample(5, random_state=42))

Porcentaje de juegos clasificados como Indie: 71.73%
--------------------------------------------------
Muestra de verificación:
                                name  \
10506                  Sparky's Hunt   
26313                        Endzeit   
2622                 Richard & Alice   
1489   The Journey Down: Chapter One   
19949       In The Long Run The Game   

                                                  genres  is_indie  
10506                                              Indie      True  
26313                                Action;Early Access     False  
2622                                     Adventure;Indie      True  
1489                                     Adventure;Indie      True  
19949  Action;Adventure;Indie;Simulation;Strategy;Ear...      True  


### 📊 Interpretación Económica e Implicancias del Negocio

Desde el punto de vista de la **economía industrial** y con el apoyo de la barrera de entrada, la variable `is_indie` representa un criterio de segmentacion para la comprecion de la estructura de la oferta. En los resultados del codigo revela un dato critico: el **$71.33\%$ de los videojuegos disponibles en la plataforma pertenecen al sector independiente**.

La abrumadora mayoria de los titulos *Indie* (reflejada por los proyectos como son *Sparky's Hunt* o *Richard & Alice*) demuestra un mercado con baja concentración en la fase de desarrollo. Impulsado por la democratización de los motores gráficos y las facilidades de autopublicación de Steam. La dinamica configura el escenario clasico de la **competencia monopolística**, en la cual miles de receptores que compiten mediante la diferenciacion de productos (como son los ormatos de juego únicos, narrativas de nicho y temáticas específicas) en lugar de hacerlo por economías de escala.

No obstante, el alto volumento de lazamientos surgiere la paradoja comercial. El ingreso masivo de los titulos independientes genera el fenomeno de saturacion e la oferta donde tiende a precionar la baja de los precios de la plataforma. Mientras las grandes corporaciones editorias (*publishers*) someten los precios altos basandose en sus costosas estrategicas de marketing. El ecosistema *Indie* se ve obligado a competir en un entorno altamente atomizado, enfrentando una demanda con alta elasticidad-precio y una feroz disputa por la visibilidad en la tienda digital.

## 2.C. Creación de Variables Temporales por Etapas Históricas

**Problema identificado:** El analisis de los lanzamientos anuales donde fragmentar el analisis, ocultanto las tendencias mecro con el objetivo de entender la evolucion de la industria. Con ello hay que agrupar en periodos de tiempo donde reflejen hitos tecnologico y comercial.

**Solución aplicada:** En primer lugar hay que asegurar de extraer los años de lazamientos de la variable `release_date`. Utilizando la funcion `pd.cut()` para la caracterizacion de los videojuegos en una de las tres etapas hitoricas:

* **Online temprano (2000-2006):** Los inicios de Steam, la transición del formato físico al digital y la consolidación de las conexiones de banda ancha.
* **HD / Casual (2006-2013):** Expansión masiva del mercado, popularización de los gráficos en alta definición y la llegada del juego casual/indie a gran escala.
* **Streaming / Esports (2013-2019):** Maduración del ecosistema, explosión de los juegos competitivos, transmisiones en vivo (Twitch) y consolidación del modelo de negocio de acceso abierto con microtransacciones.

**Justificación técnica y económica:**
* Realizar un **análisis evolutivo** agregando métricas por bloques históricos.
* Detectar **cambios estructurales** en la oferta (cantidad de lanzamientos) y en la demanda (volumen de jugadores).
* Permitir una **comparación histórica** robusta sobre cómo varían los precios promedio y los ingresos según la época en que se lanzó el juego.

In [ ]:
# Con el apoyo de la IA para la correccion de errores presentados.
# 1. Definir como datetime en el 'release_date' para extraer años.
df_steam['release_year'] = pd.to_datetime(df_steam['release_date']).dt.year

# 2. Hay que definir los grupos correspodientes de cada etapa historica.
# Los años 1999 y 2019 son los extremos que los incluyen.
limites_años = [1999, 2006, 2013, 2019]
nombres_etapas = ['Online temprano', 'HD/Casual','Streaming/Esports']

# 3. Se crea la variable con pd.cut
df_steam['historical_stage'] = pd.cut(df_steam['release_year'], bins=limites_años, labels=nombres_etapas)

# 4. Forma de verificacion que cuenta la cantidad de juegos en cada etapa
print("Distribución de videojuegos por etapa histórica:")
print(df_steam['historical_stage'].value_counts().sort_index())

Distribución de videojuegos por etapa histórica:
historical_stage
Online temprano         70
HD/Casual             1758
Streaming/Esports    25243
Name: count, dtype: int64


### 📊 Interpretación Económica e Implicancias del Negocio

Desde la perspectiva de la **economía de la innovación** y los **ciclos de mercado**,  los resultados adquiridos con la presicion del impacto de las disrupciones tecnológicas sobre la estructura de la oferta. La distribucion sectorial en las etapas historicas desciptas anteriormente revela una aceleración hiperbólica en el volumen de lanzamientos: el catálogo pasa de apenas **70 títulos** en la era del *Online temprano* a **1,758** en *HD/Casual*, para finalmente explotar de forma exponencial alcanzando **25,243 videojuegos** en la etapa de *Streaming/Esports*.

Los mercados de bienes basados en la informacion, los costos marginales de distribución tienden a cero, lo que explica esta masificación de la oferta. Aun asi, la evidencia se ve que el 93% de los títulos del dataset se concentren en la última era (*Streaming/Esports*) confirmando el escenario de **saturación de mercado**. Aunque esta tecnologia ha facilitado la democratización histórica en el acceso a los canales de publicación, las dinámicas comerciales se rigen bajo el modelo de la **Larga Cola** (*Long Tail*).

En el entorno hipercompetitivo, el ciclo de vido de estos productos se acorta drasticamente. El analisis subsiguiente de los precios promedio y de la variable `owners_mid` mediante las eras permite verificar la hipotesis central del negocio: el crecimiento expodencia de la oferta ha atomizado los ingresos de la plataforma, forzando la gran masa de nuevos competidores subsistiendo en los margenes de la Larga cola, mientras un puñado crítico de éxitos masivos retiene el poder de mercado y los retornos económicos principales.

## 2.D. Transformación Logarítmica de las Ventas Estimadas (Log-Transform)

**Problema identificado:** La distribucion de `owners_mid` presenta el sesgo positivo. La mayoria de los titulos que acumula el volumen de usuarios donde un grupo de "bestsellers" concentra millones de copias. La simetria extrema dificulta la observacion de datos y distorsiona el comportamiento a causa de los valores atipicos (*outliers*).

**Solución aplicada:** La aplicacion de la tranformacion logaritmica utilizando un logaritmo natural en la variable mas uno, mediante:


$$\log\_owners = \log(owners\_mid + 1)$$

Sumar una unidad ($+1$) es una práctica matemática estándar que previene errores de indeterminación en el caso hipotético de que existan registros con valor cero, dado que $\log(0)$ no está definido, mientras que $\log(1) = 0$.

**Justificación técnica y económica:** Esta normalización de los datos es fundamental para:
* **Evitar la dominancia de outliers:** Suaviza los valores extremos de los "hits" comerciales para que no sesguen los estadísticos descriptivos globales.
* **Lograr gráficos más legibles:** Comprime las escalas visuales, permitiendo observar la dispersión y los patrones de los juegos pequeños y medianos que antes quedaban ocultos.
* **Estabilizar las correlaciones:** Acerca la distribución de la variable a una forma normal (campana de Gauss), un supuesto crítico para realizar regresiones lineales estables y calcular coeficientes de correlación confiables.

In [ ]:
# Con el apoyo de la IA para la correccion de errores presentados.
# 1. Aplicacion de la transformacion usando la función optimizada np.log1p
# np.log1p(x) calcula exactamente el logaritmo natural de (x + 1)
df_steam['log_owners'] = np.log1p(df_steam['owners_mid'])

# 2. Verificación con el fin de comprovar las estadisticas de ambas variables
print("Comparativa de estadísticas (Original vs Transformada):")
print("-" * 60)
print(df_steam[['owners_mid', 'log_owners']].describe().round(2))

Comparativa de estadísticas (Original vs Transformada):
------------------------------------------------------------
         owners_mid  log_owners
count  2.707500e+04    27075.00
mean   1.340905e+05        9.98
std    1.328089e+06        1.34
min    1.000000e+04        9.21
25%    1.000000e+04        9.21
50%    1.000000e+04        9.21
75%    3.500000e+04       10.46
max    1.500000e+08       18.83


### 📊 Interpretación Económica e Implicancias del Negocio

Con la perspectiva del análisis estadístico de mercados, con la comparacion de las metricas descriptivas aporta la evidencia latente de una **economía de "ganadores-se-llevan-todo"** (*Winners-Take-All*), característica intrínseca de los mercados de bienes digitales.

El analisis de los datos reales, la variable original `owners_mid` muestra la asimetria extrema: con la mediana (50%) y el percentil 25% se encuentran anclados en el mínimo de **10,000 usuarios**, el valor máximo escala de forma abrupta hasta los **150,000,000 de usuarios** (un outlier masivo).  La dispersion es severa que la desviación estándar ($1.32 \times 10^6$) supera por un orden de magnitud a la media ($1.34 \times 10^5$), lo que demuestra que los estadísticos tradicionales quedan completamente distorsionados por un puñado de éxitos comerciales.

La aplicación del logaritmo natural mediante `log_owners` logra estabilizar esta variabilidad de manera exitosa, donde comprime el rango total a una escala matemáticamente manejable que va de **9.21 a 18.83**, con una media armónica de **9.98** y una desviación estándar controlada de **1.34**.

En los terminos microeconómicos, el paso metodologico se tranforma en algo crucial. La transformacion de la variable de volumen a una escala logarítmica, cambiamos la naturaleza de las futuras estimaciones. En las secciones del modelo estadistico donde la regresion lineal que vincula el precio con `log_owners` no medirá variaciones en unidades absolutas de jugadores, las cuales carecen de sentido elástico debido a la brecha entre el piso de 10,000 y el techo de 150 millones. Esto permite calcular de forma directa la **elasticidad de la demanda**. Es decir, nos dotará de la capacidad de interpretar los impactos de las estrategias de precios en **variaciones porcentuales**, permitiendo un análisis predictivo robusto sobre el comportamiento del consumidor en la plataforma.

## 2.E. Cálculo del Revenue Estimado (Ingreso Bruto Total)

**Problema identificado:** La cantidad de usuarios puede dar la nocion del alcance que tiene un juego aunque no refleja el exito financiero. Un juego gratuito, que genera un ingreso directo de cero, que puede generar millores de usuarios puede ser exitoso al igual que un juego de 60 dolares.

**Solución aplicada:** Multiplicamos el punto medio de poseedores por el precio de lista del videojuego para generar la variable cuantitativa continua `estimated_revenue`:

$$estimated\_revenue = owners\_mid \times price$$

**Justificación técnica y económica:** Esta variable es el pilar fundamental del análisis de negocio en este TP porque permite:
* Evaluar el **desempeño comercial** agregado de los diferentes actores (comparar ingresos totales de firmas Indie vs. corporaciones).
* Servir de base para modelos de **optimización**, buscando qué combinación de variables maximiza el ingreso.
* Desarrollar un **análisis económico** profundo sobre cómo se distribuyen los recursos financieros dentro de la industria de plataformas digitales.

In [ ]:
# Con el apoyo de la IA para la correccion de errores presentados.
# 1. Calculo para el Revenue Estimado multiplicando la cantidad de usuarios por el precio
df_steam['estimated_revenue'] = df_steam['owners_mid'] * df_steam['price']

# 2. Identificar el Top 5 de juegos con mayor recaudación estimada para verificar el resultado con el formateo en miles de dolares
top_revenue = df_steam[['name', 'price', 'owners_mid', 'estimated_revenue']].sort_values(by='estimated_revenue', ascending=False).head(5)

print("Top 5 videojuegos con mayor Revenue Estimado en Steam:")
print("-" * 75)
for index, row in top_revenue.iterrows():
    print(f"Juego: {row['name'][:30]:<30} | Precio: ${row['price']:<5} | Dueños: {int(row['owners_mid']):<10} | Revenue: ${row['estimated_revenue']/1e6:.2f} M")

Top 5 videojuegos con mayor Revenue Estimado en Steam:
---------------------------------------------------------------------------
Juego: PLAYERUNKNOWN'S BATTLEGROUNDS  | Precio: $26.99 | Dueños: 75000000   | Revenue: $2024.25 M
Juego: MONSTER HUNTER: WORLD          | Precio: $49.99 | Dueños: 7500000    | Revenue: $374.93 M
Juego: Grand Theft Auto V             | Precio: $24.99 | Dueños: 15000000   | Revenue: $374.85 M
Juego: HITMAN™ 2                      | Precio: $44.99 | Dueños: 7500000    | Revenue: $337.43 M
Juego: ARK: Survival Evolved          | Precio: $44.99 | Dueños: 7500000    | Revenue: $337.43 M


### 📊 Interpretación Económica e Implicancias del Negocio

Los terminos de **economía de la empresa**,  la variable `estimated_revenue` representa la aproximación al **Ingreso Total ($IT = P \times Q$)** percibido por las firmas dentro de la plataforma. Con el codigo donde aporta los datos sobre las diferentes estrategias de fijación de precios (*pricing*) pueden equilibrar los resultados financieros finales.

Con el top 5, se puede observar un fenomeno que es liderado de forma aislada por *PLAYERUNKNOWN'S BATTLEGROUNDS (PUBG)*, con un revenue estimado que rompe la escala alcanzando los **$2,024.25 millones de dólares**. En cambio, la dinamica interesante para la gestion se aprecio a la comparacion de los titulos restantes: *Grand Theft Auto V* cuenta con el doble de usuarios acumulados (**15,000,000**) que *Monster Hunter: World* o *HITMAN™ 2* (**7,500,000**); debido al precio de lista de *GTA V* ($24.99) es prácticamente la mitad que el de sus competidores ($49.99 y $44.99 respectivamente), los ingresos brutos del mismo terminan convergiendo en un umbral sumamente similar (entre los **$337M y $374M de dólares**). El hallazgo demuestra la elasticidad de la demanda y la estructura de precios determinan el posicionamiento financiero, independientemente del volumen neto de la comunidad de usuarios.

**Nota metodológica del equipo:** Hay que destacar el indicador que representa un *proxy* del ingreso bruto potencial a precios de lista. Las organizaciones del e-commerce moderno donde el valor es afectado por la baja de la estacionalidad de las liquidaciones de Steam (descuentos agresivos) y el arbitraje de precios regionales. Cabe señanar la base de datos registrados del alto perfil bajo el modelo *Premium* (pago por copia), dejando fuera los flujos de caja derivados de microtransacciones internas (*in-game purchases*), un factor crítico en modelos de negocio contemporáneos como el de *PUBG*.

Por eso, los sesgos asumidos, `estimated_revenue` se consolida como la métrica de optimización más robusta del dataset para desentrañar las asimetrías competitivas, analizar qué sectores capturan la mayor porción del excedente del consumidor y estudiar las barreras financieras que enfrentan las firmas menores ante los líderes del mercado.

## 2.F. Clasificación y Estratificación de Publishers (Editores)

**Problema identificado:** La columna `publisher` contiene miles de nombres de empresas distintas, lo que dificulta un análisis comparativo directo. Para entender la estructura competitiva del mercado, necesitamos agrupar a estos editores en categorías lógicas que reflejen su peso e influencia económica en la plataforma.

**Solución aplicada:** Primero, agrupamos el dataset por `publisher` para calcular sus métricas agregadas: la cantidad total de títulos lanzados, su promedio de usuarios (`owners_mid`) y su facturación acumulada (`estimated_revenue`).

Luego, utilizando una lógica de segmentación basada en el volumen de ingresos totales acumulados (o en su defecto, combinando volumen de catálogo y ventas), clasificamos a cada publisher en uno de tres niveles estratégicos:
* **Grandes estudios (Top / Major):** Firmas con catálogos masivos o éxitos comerciales de altísimo impacto que concentran la mayor parte del revenue del mercado.
* **Medianos (AA / Mid-tier):** Empresas consolidadas con un flujo estable de lanzamientos y ventas moderadas.
* **Indies / Pequeños:** Editores emergentes o desarrolladores independientes autogestionados con estructuras de mercado de menor escala.

**Justificación técnica y económica:** Esta estratificación empresarial es fundamental para:
* Describir la **estructura de mercado** (evaluar si es un oligopolio donde pocos publishers concentran todo el ingreso).
* Analizar las **estrategias de cartera de productos** según el tamaño de la firma.
* Facilitar la **comparación histórica** del rendimiento comercial de los pequeños competidores frente a las corporaciones dominantes.

In [26]:
# Con el apoyo de la IA para la correccion de errores presentados.
# 1. La agrupacion por publisher para calcular el perfil económico
publisher_stats = df_steam.groupby('publisher').agg(
    total_games=('name', 'count'),
    avg_owners=('owners_mid', 'mean'),
    total_revenue=('estimated_revenue', 'sum')
).reset_index()

# 2. Clasificacion usando pd.qcut y .apply()
def segmentar_publishers_por_cuantil(df):
    """Segmenta los publishers en categorías (Pequeño, Mediano, Grande)

    basándose en cuantiles del 'total_revenue'.
    """
    df = df.copy()
    labels = ["Indie/Pequeño", "Mediano", "Grande"]
    # pd.qcut with duplicates="drop" will reduce the number of bins if there are not enough unique values.
    try:
        df["categoria_publisher"] = pd.qcut(
            df["total_revenue"], q=3, labels=labels, duplicates="drop"
        )
    except ValueError as e:
        # Fallback for edge cases where 3 quantiles cannot be formed (e.g., too few unique revenue values)
        print(f"Warning: Could not create 3 quantiles for publisher segmentation: {e}. Assigning all to 'Indie/Pequeño'.")
        df["categoria_publisher"] = 'Indie/Pequeño' # Fallback for edge cases
    return df

# Aplicacion de la función al dataframe de publishers
# The function `segmentar_publishers_por_cuantil` returns a DataFrame with the new column 'categoria_publisher'.
publisher_stats = segmentar_publishers_por_cuantil(publisher_stats)

# 3. Mapear la clasificación de vuelta al dataframe principal de videojuegos
# Merge using the correctly named column 'categoria_publisher' from publisher_stats
df_steam = df_steam.merge(publisher_stats[['publisher', 'categoria_publisher']], on='publisher', how='left')

# En caso de que algún juego no tuviera publisher registrado, rellenamos como Indie/Pequeño
# Assign the 'categoria_publisher' column to 'publisher_size' in df_steam for consistency with the prompt's desired output name.
df_steam['publisher_size'] = df_steam['categoria_publisher'].fillna('Indie/Pequeño')

# 4. Verificación del resultado de la segmentación
print("Distribución de videojuegos según el tamaño de su Publisher:")
print("-" * 60)
# Print the value counts of the newly created 'publisher_size' column
print(df_steam['publisher_size'].value_counts())

Distribución de videojuegos según el tamaño de su Publisher:
------------------------------------------------------------
publisher_size
Grande           15849
Mediano           5972
Indie/Pequeño     5254
Name: count, dtype: int64


### 📊 Interpretación Económica e Implicancias del Negocio

Con la incorporacion de la variable categórica `publisher_size` introduce en el estudio de la **concentración de mercado** donde las asimetrias en el mercado de las distribuidoras. La economia de la plataforma tiene a coexistencia de miles de competidores atómicos junto a gigantes corporativos da lugar a estructuras industriales altamente polarizadas.

Cruzando las variables del volumen del catalogo (`total_games`) y desempeño comercial acumulado (`total_revenue`), donde la segmentación generada por el código aporta la base empírica para evaluar la intensidad competitiva de Steam. Esta estratificación permite aplicar en las secciones posteriores del trabajo práctico herramientas analíticas avanzadas de la economía industrial, tales como el *Coeficiente de Concentración de las g Firmas ($CR_g$)* o el *Índice de Herfindahl-Hirschman (IHH)*.

Analizar el mercado a través de este filtro tricotómico (Grandes, Medianos e Indies) es fundamental para contrastar una de las hipótesis organizacionales más relevantes del proyecto: si la proliferación masiva de títulos en la categoría *Indie/Pequeño* logra disputar de manera efectiva la generación de valor y el excedente del consumidor, o si, por el contrario, el ecosistema se rige bajo una estructura marcadamente oligopólica.

En este último escenario, un porcentaje mínimo de *Grandes Estudios* capturaría el grueso del ingreso total de la plataforma mediante economías de escala y reconocimiento de marca, relegando a la masa de pequeños creadores a competir por los rendimientos marginales en la "larga cola" (*Long Tail*) comercial.